### PR TITLE
Populate mentions tab with message history

### PR DESCRIPTION
**Pull request by 2547techno**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

An implementation of #4474 

Ended up just adding to mentions tab when message history is loaded from joining a new channel rather than just on startup.
Messages preserve their chronological order in the mentions tab, so opening a channel that you have a mention in will not necessary place that message at the bottom of the split.

### CODE REVIEW

**Comment by 2547techno** [comment]
build failed because dependency installation timed out

File: src/providers/twitch/TwitchChannel.cpp [hunk 1]
```diff
@@ -1121,6 +1121,21 @@ void TwitchChannel::loadRecentMessages()
...
+                    msg->flags.has(MessageFlag::Highlighted);
+                const auto showInMentions =
+                    msg->flags.has(MessageFlag::ShowInMentions);
+                if (highlighted && showInMentions)
+                {
+                    msgs.push_back(msg);
+                }
+            }
+
+            getApp()->getTwitch()->mentionsChannel->fillInMissingMessages(msgs);
```
Comment (Felanbird) [inline src/providers/twitch/TwitchChannel.cpp:1138]:
```suggestion
            getApp()->twitch->mentionsChannel->fillInMissingMessages(msgs);
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
 - Minor: 7TV badges now automatically update upon changing. (#4512)
 - Minor: Stream status requests are now batched. (#4713)
 - Minor: Added `/c2-theme-autoreload` command to automatically reload a custom theme. This is useful for when you're developing your own theme. (#4718)
+- Minor: Populate mentions tab with messages from history. (#4475)
```
Comment (Felanbird) [inline CHANGELOG.md:21]:
```suggestion
- Minor: Mentions loaded from message history will now appear in the /mentions tab. (#4475)
```
explains what the feature actually does better for users

Comment (Mm2PL) [reply CHANGELOG.md:21]: Maybe replace message history with historical messages

**Review by Felanbird — APPROVED** [review]
Functionality seems to work as expected (after the above suggestion is applied) 👍

**Comment by pajlada** [comment]
Closing in favour of https://github.com/Chatterino/chatterino2/pull/4752

**Pull request closed without merging**